### PR TITLE
 ✨ Update go version in kubebuilder

### PR DESCRIPTION
**Pull request by sujil02**
Updates go version to 1.16 in kubebuilder. Reference: #2067 

Signed-off-by: Sujil02 <sujil.shah@gmail.com>

### CODE REVIEW

**Review by estroz — APPROVED** [review]
/approve
/lgtm

Would like 1 more review to make sure I have not missed some needed change

/cc @Adirio 
/hold

**Comment by estroz** [comment]
@sujil02 you also need to update the version in workflow files in `.github/workflows`.

As a follow up we need to update prow configs.

**Comment by sujil02** [comment]
> .github/workflows

Done.

File: docs/book/src/migration/v2vsv3.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ and [kb-releases][kb-releases] release notes.
 
 ## Common changes
 
-v3 projects use Go modules and request Go 1.15+. Dep is no longer supported for dependency management.
+v3 projects use Go modules and request Go 1.16+. Dep is no longer supported for dependency management.
```
Comment (estroz) [inline docs/book/src/migration/v2vsv3.md:12]: Users should still be able to use go 1.15
```suggestion
v3 projects use Go modules and request Go 1.15+. Dep is no longer supported for dependency management.
```

**Comment by estroz** [comment]
If any objections come up we can consider reverting this

/lgtm
/hold cancel

**Pull request merged**